mobile/lib/modules/login/providers/authentication.provider.dart

```diff
@@ -1,23 +1,23 @@
-import 'package:dio/dio.dart';
 import 'package:flutter/material.dart';
 import 'package:hive/hive.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:immich_mobile/constants/hive_box.dart';
 import 'package:immich_mobile/modules/login/models/authentication_state.model.dart';
 import 'package:immich_mobile/modules/login/models/hive_saved_login_info.model.dart';
-import 'package:immich_mobile/modules/login/models/login_response.model.dart';
 import 'package:immich_mobile/modules/backup/services/backup.service.dart';
+import 'package:immich_mobile/shared/services/api.service.dart';
 import 'package:immich_mobile/shared/services/device_info.service.dart';
-import 'package:immich_mobile/shared/services/network.service.dart';
-import 'package:immich_mobile/shared/models/device_info.model.dart';
+import 'package:openapi/api.dart';
 
 class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
   AuthenticationNotifier(
-      this._deviceInfoService, this._backupService, this._networkService)
-      : super(
+    this._deviceInfoService,
+    this._backupService,
+    this._apiService,
+  ) : super(
           AuthenticationState(
```

Comment:
Maybe it would be cleaner to add a constructor without parameters to `AuthenticationState` that initializes the object with those default values.

Reply:
I would love to learn how to do that, I am not sure which technique should I use here

Reply:
authentication.provider.dart:
```dart
class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
  AuthenticationNotifier(
      this._deviceInfoService, this._backupService, this._networkService)
      : super(
          AuthenticationState(),
        );
```
authentication_state.model.dart:

```dart
class AuthenticationState {
  final String deviceId;
  final String deviceType;
  final String userId;
  final String userEmail;
  final bool isAuthenticated;
  final String firstName;
  final String lastName;
  final bool isAdmin;
  final bool shouldChangePassword;
  final String profileImagePath;
  final DeviceInfoRemote deviceInfo;

  AuthenticationState({
    this.deviceId = "",
    this.deviceType = "",
    this.userId = "",
    this.userEmail = "",
    this.isAuthenticated = false,
    this.firstName = "",
    this.lastName = "",
    this.isAdmin = false,
    this.shouldChangePassword = false,
    this.profileImagePath = "",
    this.deviceInfo = const DeviceInfoRemote(
      id: 0,
      userId: "",
      deviceId: "",
      deviceType: "",
      notificationToken: "",
      createdAt: "",
      isAutoBackup: false,
    ),
  });
```

This way the model class defines its own default state which I think can prevent problems in the future.
The same thing could be done for the constructor of `DeviceInfoRemote` (which also has to be const)